web/modules/custom/bnf/src/Services/BnfImporter.php

```diff
@@ -264,101 +67,40 @@ public function loadData(string $uuid, string $endpointUrl, string $nodeType = '
-
-    $response = $this->httpClient->request('post', $endpointUrl, [
-      'headers' => [
-        'Content-Type' => 'application/json',
-      ],
-      'auth' => ['bnf_graphql', getenv('BNF_GRAPHQL_CONSUMER_USER_PASSWORD')],
-      'json' => [
-        'query' => $query,
-      ],
-      // Make sure that the server is HTTPS.
-      'verify' => TRUE,
-    ]);
-
-    $data = json_decode($response->getBody()->getContents(), TRUE);
-    $nodeData = $data['data'] ?? NULL;
-
-    if (empty($nodeData)) {
-      $this->logger->error(
-        'Could not find any node data in GraphQL response. @query',
-        ['@query' => $query]
-      );
-
-      throw new \Exception('Could not retrieve content values.');
-    }
-    return $nodeData;
-
-  }
-
-  /**
-   * Importing a node from a GraphQL source endpoint.
-   */
-  public function importNode(string $uuid, string $endpointUrl, string $nodeType = 'article'): NodeInterface {
-    $nodeStorage = $this->entityTypeManager->getStorage('node');
     try {
-      $data = $this->loadData($uuid, $endpointUrl, $nodeType);
-      // Find node data, e.g. nodeArticle, nodePage etc.
-      $nodeDataKeys = preg_grep('/^node[A-Z]/', array_keys($data));
-      $nodeDataKey = $nodeDataKeys ? reset($nodeDataKeys) : [];
-      $nodeData = $data[$nodeDataKey] ?? [];
+      $endpointConfig = new SailorEndpointConfig(MangleUrl::server($endpointUrl));
+      Configuration::setEndpointFor(GetNode::class, $endpointConfig);
+      $response = GetNode::execute($uuid);
 
-      $baseUrl = $data['info']['url'] ?? NULL;
-      $nodePath = $nodeData['path'] ?? NULL;
-      $canonicalUrl = $nodeData['canonicalUrl']['url'] ?? NULL;
-      $canonicalTitle = $nodeData['canonicalUrl']['title'] ?? NULL;
+      $nodeData = $response->data?->node;
 
-      // If no custom canonical URL is set, we'll set the context of where
-      // we have exported the data from.
-      if (empty($canonicalUrl) && $baseUrl && $nodePath) {
-        $canonicalUrl = "$baseUrl$nodePath";
-        $canonicalTitle = $data['info']['name'] ?? NULL;
+      if (!$nodeData) {
+        throw new \RuntimeException('Could not fetch content.');
       }
 
-      $nodeCreation = [
-        'status' => NodeInterface::NOT_PUBLISHED,
-        'type' => $nodeType,
-        'uuid' => $uuid,
-        'title' => $nodeData['title'],
-        'field_paragraphs' => $this->getParagraphs($nodeData),
-        'field_canonical_url' => [
-          'uri' => $canonicalUrl,
-          'title' => $canonicalTitle,
-        ],
-      ];
-
-      /** @var \Drupal\node\NodeInterface $node */
-      $node = $nodeStorage->create($nodeCreation);
-      $node->save();
+      $node = $this->mapperManager->map($nodeData);
 
       $node->set(BnfStateEnum::FIELD_NAME, BnfStateEnum::Imported->value);
+
+      $node->set('status', NodeInterface::NOT_PUBLISHED);
+
+      // @todo Deal with canonical URL.
+      // CMS -> BNF: Set to site URL.
+      // BNF -> CMS: Copy whatever BNF says.
```

Comment:
I've previously "dealt" with Canonical URLs, and closed it as part of [DDFHER-187](https://reload.atlassian.net/browse/DDFHER-187)

So, either a new jira ticket has to be created, or, this todo needs to be mended as part of this pull request